File: 2640. Find the Score of All Prefixes of an Array/Solution.java
class Solution {
    public long[] findPrefixScore(int[] nums) {
        int n = nums.length, high = nums[0];
        long[] res = new long[n];
        res[0] = 1L * high + 1L * nums[0];
        for (int i = 1; i < n; i++) {
            high = Math.max(high, nums[i]);
            long conv = 1L * high + 1L * nums[i];
            res[i] = conv + res[i - 1];
        }
        return res;
    }
}

File: 2640. Find the Score of All Prefixes of an Array/Solution.py
class Solution:
    def findPrefixScore(self, nums: List[int]) -> List[int]:
        n, high = len(nums), nums[0]
        res = [0 for i in range(n)]
        res[0] = nums[0] + high
        for i in range(1, n):
            high = max(high, nums[i])
            conv = high + nums[i]
            res[i] = res[i - 1] + conv
        return res